blob/blob.go

```diff
@@ -487,9 +487,7 @@ func newBucket(b driver.Bucket) *Bucket {
 // Provider-specific types that are intended to be mutable will be exposed
 // as a pointer to the underlying type.
 //
-// See
-// https://github.com/google/go-cloud/blob/master/internal/docs/design.md#as
-// for more background.
+// See https://godoc.org/gocloud.dev for more background.
```

Comment:
Not adding an anchor because it's long and may change. I'm considering, in a followup CL, to move these "As" blocks to the main doc.go too because they're all repetetive, and then point to an #As anchor there. LMK if this makes more sense to do in one go now

Reply:
I support moving some form of this large comment to the existing section in main.go and removing it from here, and also adding a short anchor so it can be referenced.

I think the repeated comment for each `As` function should say something like:

```
// As converts i to provider-specific types. See https://godoc.org/gocloud.dev#As for more
// background, the "As" examples in this package for examples, and the provider-specific
// package documentation for the specific types supported for that provider.
```

Reply:
Done. PTAL